taskt/UI/Forms/Supplement Forms/frmHTMLElementRecorder.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Microsoft.Office.Interop.Outlook;
+using OpenQA.Selenium.Chrome;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Diagnostics;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Automation;
+using System.Windows.Forms;
+using taskt.Core.Automation.Commands;
+using taskt.Core.Automation.User32;
+
+namespace taskt.UI.Forms.Supplemental
+{
+    public partial class frmHTMLElementRecorder : UIForm
+    {
+        public frmHTMLElementRecorder()
+        {
+            InitializeComponent();
+        }
+
+        public DataTable searchParameters;
+        public string LastItemClicked;
+        private void frmHTMLElementRecorder_Load(object sender, EventArgs e)
+        {
+  
+        }
+      
+        private void pbRecord_Click(object sender, EventArgs e)
+        {
+            this.TopMost = true;
+            if (!chkStopOnClick.Checked)
+            {
+                lblDescription.Text = $"Recording.  Press F2 to stop recording!";
+            }
+
+            this.searchParameters = new DataTable();
+            this.searchParameters.Columns.Add("Enabled");
+            this.searchParameters.Columns.Add("Parameter Name");
+            this.searchParameters.Columns.Add("Parameter Value");
+            this.searchParameters.TableName = DateTime.Now.ToString("UIASearchParamTable" + DateTime.Now.ToString("MMddyy.hhmmss"));
+
+            //clear all
+            searchParameters.Rows.Clear();
+
+            //start global hook and wait for left mouse down event
+            User32Functions.GlobalHook.StartEngineCancellationHook(Keys.F2);
+            User32Functions.GlobalHook.HookStopped += GlobalHook_HookStopped;
+            User32Functions.GlobalHook.StartElementCaptureHook(chkStopOnClick.Checked);
+            webBrowser1.Document.Click += new HtmlElementEventHandler(Document_Click);
+            
+        }
+        private void GlobalHook_HookStopped(object sender, EventArgs e)
+        {
+            Document_Click(null, null);
+            this.Close();
+        }
+
+        private void Document_Click(object sender, HtmlElementEventArgs e)
+        {
+            //mouse down has occured
+            try
+            {
+                HtmlElement element = webBrowser1.Document.GetElementFromPoint(e.ClientMousePosition);
+
+                string savedId = element.Id;
+                string uniqueId = Guid.NewGuid().ToString();
+                element.Id = uniqueId;
+
+                HtmlAgilityPack.HtmlDocument doc = new HtmlAgilityPack.HtmlDocument();
+                doc.LoadHtml(element.Document.GetElementsByTagName("html")[0].OuterHtml);
+                element.Id = savedId;
+                HtmlAgilityPack.HtmlNode node = doc.GetElementbyId(uniqueId);
```

Comment:
This part's a bit confusing for me. Why are we getting the HTML element in L68 and then replacing it's ID with a unique guid ID and then changing it back? Are you doing it so that there's always a guaranteed ID for the element that we want to interact with?

Reply:
I was honestly a bit confused by that too, since I found this in the library's sample code, but yes, I believe that's pretty much what it's doing.